### PR TITLE
Use githubs gitignore files.

### PR DESCRIPTION
**Pull request by Pfeil**
I'd prefer to use this kind of gitignore file. It uses some of the well-maintained lists from https://github.com/github/gitignore. This makes basically two changes:

- Include a lot of files we have never seen and usually will not see.
- Exclude all folders named "build" by default, with the exceptions of such folders located in `/src`. If a legit exception has to be added, it can be done like this: `!/setup/build/env.sh` but to be honest I do not expect this to happen.

I left in some things I was not able to tell where it is coming from, namely `/.settings/` and `/config/`, so no lines or files were left out in the end.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,60 @@
...
-/.nb-gradle-properties
+### CUSTOM ###
+# Include directory named "build" as this name is only allowed in /src.
+#Example: !/setup/build/env.sh
+##############
+
+# Ignore other meta folders (e.g. by netbeans)
 /.settings/
 /config/
-**/application.properties
```
Comment (VolkerHartmann) [inline .gitignore:9]: I miss the following line in the new file. 
This should be also part of .gitignore as application.properties may contain private information.
(e.g. password for database, ...)
###  SPRING BOOT ###
# Ignore local settings for spring boot
# If you want to provide default settings use 'application.properties.default' and rename it while building service.
**/application.properties
##############

**Comment by Pfeil** [comment]
Sorry for the force push, I thought I did not have a pull request already. But well.

- modified the starting comment a little to make it hopefully a little better to understand what I mean.
- Added os specific files to avoid thumbnails and similar things to be committed.
- Added the line you proposed.

What do you think?

**Comment by Pfeil** [comment]
If this is fine, I think you'll have to merge, I think I can not ... probably that is why I forked the template repo back then :)

**Comment by Pfeil** [comment]
Merging the master has deleted the travis.yaml, still it seems to be triggered? The rest of the CI is probably fixed by #6 by then.

File: .gitignore [hunk 2]
```diff
@@ -1,9 +1,131 @@
...
+# and may contain slight modifications
+
+### GRADLE ###
+.gradle
+**/build/
+!src/**/build/
+# Ignore Gradle GUI config
+gradle-app.setting
+# Avoid ignoring Gradle wrapper jar file (.jar files might be ignored)
+!gradle-wrapper.jar
```
Comment (VolkerHartmann) [inline .gitignore:31]: Works not for me!?
!gradle/wrapper/gradle-wrapper.jar may be better.

**Review by VolkerHartmann — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (VolkerHartmann) [reply .gitignore:31]: Alternatively to make it more flexible:
!**/gradle-wrapper.jar

Comment (Pfeil) [reply .gitignore]: Whoops, thanks. Should be fixed now.

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1,150 @@
...
+# > backslash ("\") in front of the first "!" for patterns that begin with a
+# > literal "!", for example, "\!important!.txt".
+# From: https://git-scm.com/docs/gitignore
+############################
+
+### GRADLE ###
+!src/**/build/
+# Avoid ignoring Gradle wrapper jar file (.jar files might be ignored)
+!**/gradle-wrapper.jar
+##############
```
Comment (VolkerHartmann) [inline .gitignore:150]: New line missing at the end!

Comment (Pfeil) [reply .gitignore]: Done :)

**Pull request merged**